src/popover-v2/popover-trigger.tsx

```diff
@@ -50,10 +53,19 @@ export const PopoverTrigger = ({
         whileElementsMounted: autoUpdate,
         middleware: [
             offset(customOffset ?? 16),
-            flip(),
+            enableFlip && flip(),
             shift({
                 limiter: limitShift(),
             }),
+            enableContentScroll &&
+                size({
+                    apply({ availableHeight, elements }) {
+                        Object.assign(elements.floating.style, {
+                            maxHeight: `${availableHeight}px`,
```

Comment:
the side effect of setting overflow y on the floating wrapper is this:

<img width="312" height="295" alt="Screenshot 2025-07-25 at 10 09 24 PM" src="https://github.com/user-attachments/assets/45fd65f3-ec8a-45e3-beb7-d57054dcc1b0" />

it has to be set on the container with the border instead

Reply:
updated to set the maxHeight and overflow to the popoverCard.

cant really think of an elegant solution to pass both of them if a customPopoverContent is used instead -> user has to receive popoverResizeProps to set the maxHeight and overflow themselves

Reply:
this works! it's unavoidable that information needs to be passed down to the floating component at this point